boto3/session.py

```diff
@@ -72,6 +78,13 @@ def __repr__(self):
         return 'Session(region={0})'.format(
             repr(self._session.get_config_variable('region')))
 
+    @property
+    def profile_name(self):
+        """
+        The **read-only** profile name.
+        """
+        return self._session.profile or 'default'
```

Comment:
I *think* this probably ok at this layer.  I believe the reason it's like that in botocore is because parts of the code need to distinguish between "was explicitly set to `default` by the user" and "try the `default` profile if one hasn't been set".  The former case is an error, the latter case just means move on to the next credential provider.